offer-manager-domain/src/main/groovy/life/qbic/business/offers/create/CreateOffer.groovy

```diff
@@ -25,27 +25,35 @@ class CreateOffer implements CreateOfferInput, CalculatePrice{
 
     private CreateOfferDataSource dataSource
     private CreateOfferOutput output
+    private UpdateOffer updateOffer
 
     CreateOffer(CreateOfferDataSource dataSource, CreateOfferOutput output){
         this.dataSource = dataSource
         this.output = output
+        updateOffer = new UpdateOffer(dataSource,output)
     }
 
     @Override
     void createOffer(Offer offerContent) {
-        OfferId newOfferId = generateQuotationID(offerContent.customer)
 
-        life.qbic.business.offers.Offer finalizedOffer = new life.qbic.business.offers.Offer.Builder(
-                offerContent.customer,
-                offerContent.projectManager,
-                offerContent.projectTitle,
-                offerContent.projectDescription,
-                offerContent.items,
-                offerContent.selectedCustomerAffiliation)
-                .identifier(newOfferId)
-                .build()
+        if(offerContent.identifier == null){
+            OfferId newOfferId = generateQuotationID(offerContent.customer)
 
-        storeOffer(finalizedOffer)
+            life.qbic.business.offers.Offer finalizedOffer = new life.qbic.business.offers.Offer.Builder(
+                    offerContent.customer,
+                    offerContent.projectManager,
+                    offerContent.projectTitle,
+                    offerContent.projectDescription,
+                    offerContent.items,
+                    offerContent.selectedCustomerAffiliation)
+                    .identifier(newOfferId)
+                    .build()
+
+            storeOffer(finalizedOffer)
+
+        }else{
+            updateOffer.updateOffer(offerContent)
+        }
```

Comment:
I think it is cleaner to read, when we remove the huge indentation blocks, for example like:

```suggestion
            if (offer.identifier) {
                updateOffer.updateOffer(offerContent)
            } else {
                createNewOffer(offerContent)
            }
```

Reply:
Nice, I like your suggestion, I applied and committed it